plugins/balloontoolbar/plugin.js

```diff
@@ -680,6 +680,32 @@
 				this._detachListeners();
 			};
 
+			/**
+			 * Updates status of passed element.
+			 *
+			 * @since 4.9.0
+			 * @private
+			 * @param {CKEDITOR.ui.button/CKEDITOR.ui.richCombo}
+			 * @member CKEDITOR.ui.balloonToolbarView
+			 */
+			CKEDITOR.ui.balloonToolbarView.prototype._updateStatus = function( item ) {
```

Comment:
This method is copied from `CKEDITOR.ui.richCombo`. I see two problems with it:

1. It is exact copy and if something changes in a source method (well, not probable, but may happen), it may stop working.
2. `Item` param type seems to be `CKEDITOR.ui.button` or `CKEDITOR.ui.richCombo`, however this method is copied from `CKEDITOR.ui.richCombo` so it will not work for buttons. As this method is only used for richCombo maybe it should only accept only `CKEDITOR.ui.richCombo` type?

As for **1.** my suggestion is to make `updateStatus` method in `CKEDITOR.ui.richCombo` public (or add some new method which will internally call `updateStatus`) and then you could get rid of this method entirely which will simplify the code.